### PR TITLE
Populates EnvelopeResource

### PR DESCRIPTION
**Pull request by nelsitoPuglisi**
## Goal

Populates the `envelope.resource` field, without using MetadataService.

Note: MetadataService is still used because of reactNativeBundleId which will be refactored in a subsequent PR.

### CODE REVIEW

**Comment by nelsitoPuglisi** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/516?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#516** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/516?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#515** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/515?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#514** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/514?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nelsitoPuglisi and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt [hunk 1]
```diff
@@ -1,9 +1,60 @@
...
+import io.embrace.android.embracesdk.capture.metadata.MetadataService
 import io.embrace.android.embracesdk.internal.payload.EnvelopeResource
+import io.embrace.android.embracesdk.payload.AppInfo
+import io.embrace.android.embracesdk.payload.DeviceInfo
 
-internal class EnvelopeResourceSourceImpl : EnvelopeResourceSource {
+internal class EnvelopeResourceSourceImpl(
+    private val deviceInfo: DeviceInfo,
+    private val appInfo: AppInfo,
+    private val metadataService: MetadataService,
```
Comment (fractalwrench) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt:12]: IMO we have a good opportunity to get rid of the `MetadataService` by copying its existing functions into here instead. `MetadataService` could then be deleted later on when we fully switch over to the new payload.

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/payload/EnvelopeResource.kt [hunk 2]
```diff
@@ -1,104 +1,201 @@
...
+ * @param numCores (Android) The number of CPU cores the device has. Previous name: d.nc
  */
+
 @JsonClass(generateAdapter = true)
 internal data class EnvelopeResource(
+
+    /* The app's publicly displayed version name. Previous name: a.v */
     @Json(name = "app_version")
-    val appVersion: String? = null,
+    val appVersion: kotlin.String? = null,
```
Comment (fractalwrench) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/payload/EnvelopeResource.kt:78]: Would be good to manually tweak the generated code to avoid the fully qualified name for these types (and other style issues). Could also go in a separate PR if you want to get it merged quickly

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt (hunk 1 above)
Comment (nelsitoPuglisi) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt]: Done in #546

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/payload/EnvelopeResource.kt (hunk 2 above)
Comment (nelsitoPuglisi) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/payload/EnvelopeResource.kt:78]: I reverted that manually now. I can work on the generator after this is merged.

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt [hunk 3]
```diff
@@ -1,9 +1,60 @@
...
+            diskTotalCapacity = deviceInfo.internalStorageTotalCapacity,
+            osType = deviceInfo.operatingSystemType,
+            osVersion = deviceInfo.operatingSystemVersion,
+            osCode = deviceInfo.operatingSystemVersionCode.toString(),
+            screenResolution = deviceInfo.screenResolution,
+            numCores = deviceInfo.cores,
+        )
+    }
+
+    private fun mapFramework(appFramework: AppFramework): EnvelopeResource.AppFramework {
```
Comment (fractalwrench) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt:45]: It sounds like we'll need to change this to an `Int` again

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt [hunk 4]
```diff
@@ -1,9 +1,60 @@
...
 
+import io.embrace.android.embracesdk.Embrace.AppFramework
 import io.embrace.android.embracesdk.internal.payload.EnvelopeResource
+import io.embrace.android.embracesdk.payload.AppInfo
+import io.embrace.android.embracesdk.payload.DeviceInfo
 
-internal class EnvelopeResourceSourceImpl : EnvelopeResourceSource {
+internal class EnvelopeResourceSourceImpl(
+    private val deviceInfo: DeviceInfo,
+    private val appInfo: AppInfo,
```
Comment (fractalwrench) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt:10]: Would be ideal to avoid these types as they're the old payload that will eventually be deleted. Is this addressed in another changeset?

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt [hunk 5]
```diff
@@ -1,9 +1,60 @@
...
+            appVersion = appInfo.appVersion,
+            appEcosystemId = packageName,
+            appFramework = mapFramework(appFramework),
+            buildId = appInfo.buildId,
+            buildType = appInfo.buildType,
+            buildFlavor = appInfo.buildFlavor,
+            environment = appInfo.environment,
+            bundleVersion = appInfo.bundleVersion,
+            sdkVersion = appInfo.sdkVersion,
+            sdkSimpleVersion = appInfo.sdkSimpleVersion!!.toInt(),
```
Comment (fractalwrench) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt:25]: Will this always be set? I'd feel better about using the null coalescing operator here so we don't crash if it is null for whatever reason

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/payload/EnvelopeResource.kt [hunk 6]
```diff
@@ -1,104 +1,201 @@
...
     val deviceArchitecture: String? = null,
 
+    /* Whether the device is rooted/jailbroken or not. Previous name: d.jb */
     @Json(name = "jailbroken")
     val jailbroken: Boolean? = null,
 
+    /* The total capacity of internal storage for the whole device. Previous name: d.ms */
     @Json(name = "disk_total_capacity")
-    val diskTotalCapacity: Long? = null,
+    val diskTotalCapacity: kotlin.Long? = null,
```
Comment (fractalwrench) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/payload/EnvelopeResource.kt:159]:
```suggestion
    val diskTotalCapacity: Long? = null,
```

**Review by fractalwrench — COMMENTED** [review]
Left a few comments inline - also it looks like there are some compilation issues on CI?

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt (hunk 3 above)
Comment (nelsitoPuglisi) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt:47]: haha yeah... I'm on it.

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt (hunk 5 above)
Comment (nelsitoPuglisi) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt]: Is not using appInfo anymore

**Review by lucaslabari — APPROVED** [review]
LGTM

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/payload/EnvelopeResource.kt [hunk 7]
```diff
@@ -1,104 +1,210 @@
...
     val bundleVersion: String? = null,
 
+    /* The version number of the Embrace SDK. Previous name: a.sdk */
     @Json(name = "sdk_version")
     val sdkVersion: String? = null,
 
+    /* The simple version number of the Embrace SDK. Previous name: a.sdc */
     @Json(name = "sdk_simple_version")
-    val sdkSimpleVersion: String? = null,
+    val sdkSimpleVersion: Int? = null,
```
Comment (fractalwrench) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/payload/EnvelopeResource.kt:115]: Should this really be int?

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt [hunk 8]
```diff
@@ -1,9 +1,69 @@
...
+    private val appFramework: AppFramework,
+    private val deviceArchitecture: DeviceArchitecture,
+    private val device: Device,
+    private val metadataService: MetadataService
+) : EnvelopeResourceSource {
+
     override fun getEnvelopeResource(): EnvelopeResource {
-        throw NotImplementedError("Not yet implemented")
+        return EnvelopeResource(
+            appVersion = packageInfo.versionName.toString().trim { it <= ' ' },
```
Comment (fractalwrench) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt:26]: What does this do & is it a change in behavior from the previous implementation?

**Review by fractalwrench — APPROVED** [review]
LGTM

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt (hunk 8 above)
Comment (nelsitoPuglisi) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt:26]: yes, it's the same behavior. https://github.com/embrace-io/embrace-android-sdk/blob/f1752cf7319d63e151309d57a7c56a4916744895/embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/injection/EssentialServiceModule.kt#L103

I don't fully understand yet why the trim is required

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/payload/EnvelopeResource.kt (hunk 7 above)
Comment (nelsitoPuglisi) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/payload/EnvelopeResource.kt:115]: That's what our spec determine 🤷‍♂️
